api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementDecisionFactoryTest.java

```diff
@@ -0,0 +1,140 @@
+        UUID existingClaimId = UUID.randomUUID();
+
+        //When
+        EligibilityAndEntitlementDecision eligibilityAndEntitlementDecision = factory.buildDecision(eligibilityResponse, entitlement, existingClaimId,
+                Optional.empty(), false);
+
+        //Then
+        EligibilityAndEntitlementDecision expectedDecision = aValidDecisionBuilder()
+                .identityAndEligibilityResponse(eligibilityResponse)
+                .existingClaimId(existingClaimId)
+                .hmrcHouseholdIdentifier(null)
+                .build();
+        assertThat(eligibilityAndEntitlementDecision).isEqualTo(expectedDecision);
+    }
+
+    @Test
+    void shouldBuildDecisionWithExistingClaimUUIDAndHmrcHouseholdIdentifier() {
+        //Given
+        PaymentCycleVoucherEntitlement entitlement = aPaymentCycleVoucherEntitlementWithVouchers();
+        UUID existingClaimId = UUID.randomUUID();
+
+        //When
+        EligibilityAndEntitlementDecision eligibilityAndEntitlementDecision = factory.buildDecision(eligibilityResponse, entitlement, existingClaimId,
+                Optional.of(HMRC_HOUSEHOLD_IDENTIFIER), false);
+
+        //Then
+        EligibilityAndEntitlementDecision expectedDecision = aValidDecisionBuilder()
+                .identityAndEligibilityResponse(eligibilityResponse)
+                .existingClaimId(existingClaimId)
+                .hmrcHouseholdIdentifier(HMRC_HOUSEHOLD_IDENTIFIER)
+                .build();
+        assertThat(eligibilityAndEntitlementDecision).isEqualTo(expectedDecision);
+    }
+
+    @Test
+    void shouldBuildDecisionWithoutExistingClaimUUID() {
+        //Given
+        PaymentCycleVoucherEntitlement entitlement = aPaymentCycleVoucherEntitlementWithVouchers();
+
+        //When
+        EligibilityAndEntitlementDecision eligibilityAndEntitlementDecision = factory.buildDecision(eligibilityResponse, entitlement, Optional.empty(), false);
+
+        //Then
+        EligibilityAndEntitlementDecision expectedDecision = aValidDecisionBuilder()
+                .identityAndEligibilityResponse(eligibilityResponse)
+                .hmrcHouseholdIdentifier(null)
+                .build();
+        assertThat(eligibilityAndEntitlementDecision).isEqualTo(expectedDecision);
+    }
+
+    @Test
+    void shouldBuildDecisionForDuplicateClaim() {
+        //Given
+        PaymentCycleVoucherEntitlement entitlement = aPaymentCycleVoucherEntitlementWithVouchers();
+
+        //When
+        EligibilityAndEntitlementDecision eligibilityAndEntitlementDecision = factory.buildDecision(eligibilityResponse, entitlement, Optional.empty(), true);
+
+        //Then
+        EligibilityAndEntitlementDecision expectedDecision = aValidDecisionBuilder()
+                .identityAndEligibilityResponse(eligibilityResponse)
+                .hmrcHouseholdIdentifier(null)
+                .voucherEntitlement(null)
+                .eligibilityStatus(EligibilityStatus.DUPLICATE)
```

Comment:
I think we should extract the `isDuplicate` parameter out to a variable to make it clearer. Took me a while to understand why we expected a duplicate status.

Reply:
Yes, good idea, will do.